### DISCUSSION TITLE
Can an Aspect be added to a custom derived attribute class such as "public class MyAttribute : AuthorizeAttribute"?

### DISCUSSION THREAD

**Discussion by HowardShank** [metalama/Metalama#261] [Q&A] [answered]
Can an Aspect be added to a custom derived attribute class such as "public class MyAttribute : AuthorizeAttribute"?

```
[MyAspect]
public class MyAttribute : AuthorizeAttribute
{
...
}
```
If yes, could this be a potential way to modify the behavior of MyAttribute class to dynamically provide Policy/Role data overriding the compile time parameter values requirement of the AuthorizeAttribute? I'll explore the validity of this concept if it seems plausible.

I'm just exploring concepts/ways to come up with a solution to dynamic authorization of my users without creating code blocks in each and every Blazor page.

**Comment by gfraiteur** [comment, marked as answer]
Yes, this is possible. The modifications to the attribute will be visible at run time through reflection.

Another question is whether this is desirable. Unless you have many of these attributes (which I find weird), using an aspect on an attribute could be more complex than a pure object-oriented solution.

**Reply by HowardShank** [reply]
Thank you for your response.

Now I just need to figure out how to add a property with the new modifier "string new PropertyName{get;set;}"

**Reply by gfraiteur** [reply]
You should use:

```cs
[Introduce( WhenExists = OverrideStrategy .New )]
```
